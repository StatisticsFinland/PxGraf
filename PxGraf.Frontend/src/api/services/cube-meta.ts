/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { ICubeMeta } from "types/cubeMeta";
import { defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Interface for cube metadata result. Cube is a multi-dimensional result of Px file data filtered by a query.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {ICubeMeta} data - The cube meta data as an @see {@link ICubeMeta} object.
 */
export interface ICubeMetaResult {
    isLoading: boolean;
    isError: boolean;
    data: ICubeMeta;
}

const fetchCubeMeta = async(idStack: string[]): Promise<ICubeMeta> => {
    const client = new ApiClient();
    const url = `creation/cube-meta/${idStack.join('/')}`;

    return await client.getAsync(url);
}

export const useCubeMetaQuery = (idStack: string[]): ICubeMetaResult => {
    return useQuery(
        ['cube-meta', idStack],
        () => fetchCubeMeta(idStack),
        defaultQueryOptions
    );
}