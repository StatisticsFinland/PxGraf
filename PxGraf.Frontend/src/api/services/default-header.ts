/* istanbul ignore file */

import ApiClient from "api/client";
import { merge } from "lodash";
import { useQuery } from "react-query";
import { MultiLanguageString } from "types/multiLanguageString";
import { Query } from "types/query";
import { buildTableReference, defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Interface for the chart's header result
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred while retrieving the header.
 * @property {MultiLanguageString} data - The header data represented as a multi-language string.
 */
export interface IHeaderResult {
    isLoading: boolean;
    isError: boolean;
    data: MultiLanguageString;
}

const fetchDefaultHeader = async(idStack: string[], query: Query): Promise<MultiLanguageString> => {
    const client = new ApiClient();
    const requestBody = JSON.stringify(
        merge(
            { tableReference: buildTableReference(idStack) },
            { variableQueries: query }
        )
    );
  
    const url = 'creation/default-header';
    return await client.postAsync(url, requestBody);
}

export const useDefaultHeaderQuery = (idStack: string[], query: Query): IHeaderResult => {
    return useQuery(
        ['default-header', ...idStack, query],
        () => fetchDefaultHeader(idStack, query),
        {
            ...defaultQueryOptions,
            enabled: query != null,
        }
    );
}