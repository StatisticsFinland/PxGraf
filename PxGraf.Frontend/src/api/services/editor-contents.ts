/* istanbul ignore file */

import ApiClient from "api/client";
import { useQuery } from "react-query";
import { IEditorContentsResponse } from "types/editorContentsResponse";
import { buildCubeQuery, defaultQueryOptions, useDebounceState } from "utils/ApiHelpers";
import { ICubeQuery, Query } from "types/query";

/**
 * Interface for editor contents result. Editor contents are used by the editor to populate the settings for adjusting the visualization.
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {IEditorContentsResponse} data - The editor contents as an @see {@link IEditorContentsResponse} object.
 */ 
export interface IEditorContentsResult {
    isLoading: boolean;
    isError: boolean;
    data: IEditorContentsResponse;
}

const fetchEditorContents = async (idStack: string[], query: Query, metaEdits: ICubeQuery): Promise<IEditorContentsResponse> => {
    const client = new ApiClient();
    const url = 'creation/editor-contents';
    const requestBody = JSON.stringify(buildCubeQuery(query, metaEdits, idStack));
    return await client.postAsync(url, requestBody);
}

export const useEditorContentsQuery = (idStack: string[], query: Query, cubeQuery: ICubeQuery): IEditorContentsResult => {
    [idStack, query, cubeQuery] = useDebounceState(1000, idStack, query, cubeQuery);
    const queryKey = ['editor-contents', ...idStack, query, cubeQuery];
    return useQuery(
        queryKey,
        () => fetchEditorContents(idStack, query, cubeQuery),
        {
            ...defaultQueryOptions,
            enabled: query != null,
            keepPreviousData: true,
        }
    );
}