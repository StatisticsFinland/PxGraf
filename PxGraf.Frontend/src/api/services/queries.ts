/* istanbul ignore file */

import ApiClient from "api/client";
import { useMutation, useQuery } from "react-query";
import { ICubeQuery, Query } from "types/query";
import { IVisualizationSettings } from "types/visualizationSettings";
import { buildCubeQuery, defaultQueryOptions } from "utils/ApiHelpers";

/**
 * Response object for saving a query.
 * @property {string} id - The id of the saved query.
 */
export interface ISaveQueryResponse {
    id: string;
}

/**
 * Response object for fetching a saved query.
 * @property query - The query object consisting of the table reference and variable queries @see {@link Query}.
 * @property {IVisualizationSettings} settings - The visualization settings for the query.
 */
export interface IFetchSavedQueryResponse {
    query: {
        tableReference: {
            name: string;
            hierarchy: string[];
        };
        variableQueries: Query;
    } & ICubeQuery;
    settings: IVisualizationSettings;
}

/**
 * Interface for save query result
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {boolean} isSuccess - Flag to indicate if the request was successful.
 * @property {ISaveQueryResponse} data - The save query response data.
 * @property {function} mutate - Function to mutate the state of the query.
 */
export interface ISaveQueryResult {
    isLoading: boolean;
    isError: boolean;
    isSuccess: boolean;
    data: ISaveQueryResponse;
    mutate: (property: boolean) => void;
}

/**
 * Interface for fetch saved query result
 * @property {boolean} isLoading - Flag to indicate if the data is still loading.
 * @property {boolean} isError - Flag to indicate if an error occurred during loading.
 * @property {boolean} isSuccess - Flag to indicate if the request was successful.
 * @property {IFetchSavedQueryResponse} data - The fetch saved query response data.
 */
export interface IFetchSavedQueryResult {
    isLoading: boolean;
    isError: boolean;
    isSuccess: boolean;
    data: IFetchSavedQueryResponse;
}

const sendSaveRequest = async (
    idStack: string[],
    query: Query,
    metaEdits: ICubeQuery,
    selectedVisualization: string,
    visualizationSettings: IVisualizationSettings,
    archive: boolean | void
): Promise<ISaveQueryResponse> => {

    const client = new ApiClient();

    const requestBody = JSON.stringify({
        query: buildCubeQuery(query, metaEdits, idStack),
        settings: {
            ...visualizationSettings,
            selectedVisualization: selectedVisualization,
        }
    });
    const url = archive ? 'sq/archive' : 'sq/save';
    return await client.postAsync(url, requestBody);
}

export const fetchSavedQuery = async (queryId: string): Promise<IFetchSavedQueryResponse> => {
    const client = new ApiClient();
    const url = 'sq/' + queryId;
    const response = await client.getAsync(url);

    // Check for VariableQueries and assign to variableQueries
    if (!response.query?.variableQueries && response.query?.VariableQueries) {
        response.query.variableQueries = response.query.VariableQueries;
    }

    return response as IFetchSavedQueryResponse;
};

export const useFetchSavedQuery = (queryId: string): IFetchSavedQueryResult => {
    return useQuery(
        [queryId],
        () => fetchSavedQuery(queryId),
        defaultQueryOptions
    );
};

export const useSaveMutation = (idStack: string[], query: Query, metaEdits: ICubeQuery, selectedVisualization: string, visualizationSettings: IVisualizationSettings): ISaveQueryResult => {
    return useMutation(
        (archive) => sendSaveRequest(idStack, query, metaEdits, selectedVisualization, visualizationSettings, archive),
        { retry: false }
    );
}