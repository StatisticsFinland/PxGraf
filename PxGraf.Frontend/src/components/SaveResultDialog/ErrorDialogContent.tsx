import { useTranslation } from 'react-i18next';

import {
    Alert, DialogContent,
  } from '@mui/material';
import React from 'react';

export const ErrorDialogContent: React.FC = () => {
    const { t } = useTranslation();

    return (
        <DialogContent dividers={true}>
            <Alert severity="error"> {t('saveResultDialog.errorMessage')} </Alert>
        </DialogContent>
    );
}

export default ErrorDialogContent;