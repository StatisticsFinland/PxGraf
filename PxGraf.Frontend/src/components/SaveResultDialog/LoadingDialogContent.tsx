import { CircularProgress, Container, DialogContent } from '@mui/material';
import React from 'react';
import styled from 'styled-components';

const ProgressWrapper = styled(Container)`
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const LoadingDialogContent: React.FC = () => {
    return (
        <DialogContent dividers={true}>
            <ProgressWrapper>
                <CircularProgress />
            </ProgressWrapper>
        </DialogContent>
    );
}

export default LoadingDialogContent;