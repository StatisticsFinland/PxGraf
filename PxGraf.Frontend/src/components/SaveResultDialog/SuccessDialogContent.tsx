import { useTranslation } from 'react-i18next';

import {
    DialogContent, FormControl, IconButton, InputAdornment, InputLabel, OutlinedInput
} from '@mui/material';

import ContentPasteIcon from '@mui/icons-material/ContentPaste';
import styled from 'styled-components';
import React from 'react';

const StyledOutlinedInput = styled(OutlinedInput)`
    width: 380px;
`;

interface ISuccessDialogContentProps {
    queryId?: string;
}

export const SuccessDialogContent: React.FC<ISuccessDialogContentProps> = ({ queryId }) => {

    const { t } = useTranslation();
    const text = queryId || "Unknown";

    return (
        <DialogContent dividers={true}>
            <FormControl>
                <InputLabel htmlFor="display-saved-query-id-input-label">{t("saveResultDialog.id")}</InputLabel>
                <StyledOutlinedInput
                    id="display-saved-query-id-input"
                    defaultValue={text}
                    inputProps={{ readOnly: true }}
                    label={t("saveResultDialog.id")}
                    endAdornment={
                        <InputAdornment position="end">
                            <IconButton
                                aria-label={t("saveResultDialog.copyToClipBoard")}
                                onClick={() => navigator.clipboard.writeText(text)}
                                edge="end">
                                <ContentPasteIcon />
                            </IconButton>
                        </InputAdornment>}
                />
            </FormControl>
        </DialogContent>
    );
}

export default SuccessDialogContent;