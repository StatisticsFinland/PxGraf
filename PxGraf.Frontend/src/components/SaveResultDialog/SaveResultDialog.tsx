import { useTranslation } from 'react-i18next';

import { Button, Dialog, DialogTitle, DialogActions } from '@mui/material';

import { SuccessDialogContent } from './SuccessDialogContent';
import { ErrorDialogContent } from './ErrorDialogContent';
import { LoadingDialogContent } from './LoadingDialogContent';
import React from 'react';
import { ISaveQueryResult } from 'api/services/queries';

interface ISaveResultDialogProps {
    open: boolean;
    onClose: () => void;
    mutation: ISaveQueryResult;
}

interface ISaveDialogContentProps {
    result: ISaveQueryResult;
}

export const SaveResultDialog: React.FC<ISaveResultDialogProps> = ({ open, onClose, mutation}) => {
    
    const { t } = useTranslation();
    
    const HeaderText = () => {
        if(mutation.isLoading) { return t('saveDialog.saveQuery') }
        if(mutation.isError) { return t('saveResultDialog.fail') }
        if(mutation.isSuccess) { return t('saveResultDialog.success') }
    }

    return (
        <Dialog
        open={open}
        onClose={onClose}
        scroll='paper'
        aria-labelledby="scroll-dialog-title"
        aria-describedby="scroll-dialog-description"
        >
            <DialogTitle id="result-dialog-title">{ HeaderText() }</DialogTitle>
            <SaveDialogContent result={mutation}/>
            <DialogActions>
                <Button onClick={onClose} variant="contained">{t("saveResultDialog.ok")}</Button>
            </DialogActions>
        </Dialog>
    );
}

const SaveDialogContent: React.FC<ISaveDialogContentProps> = ({result}) => {
    if(result.isLoading) { return <LoadingDialogContent/> }
    else if(result.isSuccess) { return <SuccessDialogContent queryId = {result.data?.id}/>; }
    else { return <ErrorDialogContent/> }
}

export default SaveResultDialog;