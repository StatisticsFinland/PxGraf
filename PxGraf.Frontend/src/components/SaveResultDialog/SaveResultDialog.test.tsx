import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import SaveResultDialog from './SaveResultDialog';
import { ISaveQueryResult } from 'api/services/queries';

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => null),
            },
        };
    },
}));

const mockSuccessMutation: ISaveQueryResult = {
    isLoading: false,
    isError: false,
    isSuccess: true,
    data: {id:'foobar'},
    mutate: function (_property: any): void {
        throw new Error('Function not implemented.');
    }
};

const mockErrorMutation: ISaveQueryResult = {
    isLoading: false,
    isError: false,
    isSuccess: true,
    data: {id:'foobar'},
    mutate: function (_property: any): void {
        throw new Error('Function not implemented.');
    }
};

const onCloseMock = jest.fn();

describe('Rendering test', () => {
    it('renders correctly when closed', () => {
        const dom = render(<SaveResultDialog mutation={mockSuccessMutation} onClose={onCloseMock} open={false} />);
        expect(dom.baseElement).toMatchSnapshot();
    });

    it('renders correctly when open', () => {
        const dom = render(<SaveResultDialog mutation={mockSuccessMutation} onClose={onCloseMock} open={true} />);
        expect(dom.baseElement).toMatchSnapshot();
    });
});

describe('Error rendering test', () => {
    it('renders error correctly when open', () => {
        const dom = render(<SaveResultDialog mutation={mockErrorMutation} onClose={onCloseMock} open={true} />);
        expect(dom.baseElement).toMatchSnapshot();
    });
});

describe('Assertion test', () => {
    it('invokes close function when cancel button is clicked', () => {
        render(<SaveResultDialog mutation={mockSuccessMutation} onClose={onCloseMock} open={true} />);
        fireEvent.click(screen.getByText('saveResultDialog.ok'));
        expect(onCloseMock).toBeCalledTimes(1);
    });
});