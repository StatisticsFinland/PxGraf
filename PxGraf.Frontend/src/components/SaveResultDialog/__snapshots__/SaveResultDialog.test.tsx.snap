// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error rendering test renders error correctly when open 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-1424xw8-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-1cennmq-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-19do60a-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-describedby="scroll-dialog-description"
        aria-labelledby="scroll-dialog-title"
        aria-modal="true"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-10d30g3-MuiPaper-root-MuiDialog-paper"
        role="dialog"
        style="--Paper-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-rvoa5x-MuiTypography-root-MuiDialogTitle-root"
          id="result-dialog-title"
        >
          saveResultDialog.success
        </h2>
        <div
          class="MuiDialogContent-root MuiDialogContent-dividers css-10bxh4q-MuiDialogContent-root"
        >
          <div
            class="MuiFormControl-root css-1iw3t7y-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="display-saved-query-id-input-label"
            >
              saveResultDialog.id
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd sc-Qotzb cZOGQx css-jupps9-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
                id="display-saved-query-id-input"
                readonly=""
                type="text"
                value="foobar"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"
              >
                <button
                  aria-label="saveResultDialog.copyToClipBoard"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="ContentPasteIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 2h-4.18C14.4.84 13.3 0 12 0S9.6.84 9.18 2H5c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2m-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1m7 18H5V4h2v3h10V4h2z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w1u3ce"
                >
                  <span>
                    saveResultDialog.id
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-15fu35s-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            saveResultDialog.ok
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`Rendering test renders correctly when closed 1`] = `
<body>
  <div />
</body>
`;

exports[`Rendering test renders correctly when open 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-1424xw8-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-1cennmq-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-19do60a-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-describedby="scroll-dialog-description"
        aria-labelledby="scroll-dialog-title"
        aria-modal="true"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-10d30g3-MuiPaper-root-MuiDialog-paper"
        role="dialog"
        style="--Paper-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-rvoa5x-MuiTypography-root-MuiDialogTitle-root"
          id="result-dialog-title"
        >
          saveResultDialog.success
        </h2>
        <div
          class="MuiDialogContent-root MuiDialogContent-dividers css-10bxh4q-MuiDialogContent-root"
        >
          <div
            class="MuiFormControl-root css-1iw3t7y-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="display-saved-query-id-input-label"
            >
              saveResultDialog.id
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd sc-Qotzb cZOGQx css-jupps9-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-1dune0f-MuiInputBase-input-MuiOutlinedInput-input"
                id="display-saved-query-id-input"
                readonly=""
                type="text"
                value="foobar"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-elo8k2-MuiInputAdornment-root"
              >
                <button
                  aria-label="saveResultDialog.copyToClipBoard"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1ysp02-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                    data-testid="ContentPasteIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 2h-4.18C14.4.84 13.3 0 12 0S9.6.84 9.18 2H5c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2m-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1m7 18H5V4h2v3h10V4h2z"
                    />
                  </svg>
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-w1u3ce"
                >
                  <span>
                    saveResultDialog.id
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-15fu35s-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            saveResultDialog.ok
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
